manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Connector.java

```diff
@@ -0,0 +1,155 @@
+package com.redhat.service.smartevents.manager.v2.persistence.models;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedNativeQueries;
+import javax.persistence.NamedNativeQuery;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.hibernate.annotations.Filter;
+import org.hibernate.annotations.FilterDef;
+import org.hibernate.annotations.FilterDefs;
+import org.hibernate.annotations.Filters;
+import org.hibernate.annotations.ParamDef;
+
+import com.redhat.service.smartevents.infra.core.models.connectors.ConnectorType;
+import com.redhat.service.smartevents.infra.v2.api.models.connectors.ConnectorDefinition;
+
+@NamedQueries({
+        @NamedQuery(name = "CONNECTOR_V2.findByIdWithConditions",
+                query = "from Connector_V2 conn left join fetch conn.conditions where conn.id=:id and conn.type=:type"),
+        @NamedQuery(name = "CONNECTOR_V2.findByIdsWithBridgeAndConditions",
+                query = "select distinct (conn) from Connector_V2 conn left join fetch conn.bridge left join fetch conn.conditions where conn.id in (:ids)"),
+})
+// Hibernate does not support sub-queries in Named Queries. This is therefore written as Native Query.
+// Hibernate however does not support eager fetches with Native Queries without composing a _View_ class model,
+// retrieving a flat ResultSet and then building the object model hierarchy in Java. We therefore split
+// retrieval into two database calls: (1) Get the Connector IDs, (2) Fetch the Connector objects.
+@NamedNativeQueries({
+        @NamedNativeQuery(name = "CONNECTOR_V2.findConnectorIdByShardIdToDeployOrDelete",
+                query = "select conn.id " +
+                        "from Connector_V2 conn " +
+                        "left join Bridge_V2 b on conn.bridge_id = b.id " +
+                        "left join (" +
+                        "  select connector_id, count(*) as incomplete_count from Condition c where component='MANAGER' and status != 'TRUE' group by c.connector_id " +
+                        "    ) cp on cp.connector_id = conn.id " +
+                        "where " +
+                        // The LEFT JOIN on the sub-query can return a null if there are no MANAGER records that are NOT complete.
+                        "(cp.incomplete_count = 0 or cp.incomplete_count is null) and " +
+                        "b.shard_id = :shardId and " +
+                        "conn.type = :type")
+})
+@FilterDefs({
```

Comment:
You don't have support for filters in your DAO. Did you want to keep the definitions on your Entity?

Reply:
I'll remove them, they will be added/tested in the next PRs (https://issues.redhat.com/browse/MGDOBR-1398 for example) :+1: